src/arithmetic/aggregate.c

```diff
@@ -54,5 +54,6 @@ inline void *Agg_FuncCtx(AggCtx *ctx) {
 }
 
 inline void Agg_SetResult(struct AggCtx *ctx, SIValue v) {
-	ctx->result = v;
+	ctx->result = SI_ShareValue(v);
```

Comment:
Can you move this call to the invoking function? Usually, we share the value from the value owner and not on the value receiver, or use `SIValue_MakeVolatile`
since this is a very delicate resource handling, please add a comment for the reason and meaning of making this value volatile